tests/sentry/hybridcloud/test_caching.py

```diff
@@ -158,3 +161,118 @@ def cache_death_event() -> Generator[None, None, None]:
     random = Random(84716393)
     for i in range(10000):
         next(random.choice([reader1, reader2, writer1, cache_death]))
+
+
+@django_db_all(transaction=True)
+def test_caching_many() -> None:
+    cache.clear()
+
+    @back_with_silo_cache_many(base_key="get_users", silo_mode=SiloMode.REGION, t=RpcUser)
+    def get_users(user_ids: list[int]) -> list[RpcUser]:
+        return user_service.get_many(filter=dict(user_ids=user_ids))
+
+    users = [Factories.create_user() for _ in range(3)]
+    user_ids = [u.id for u in users]
+
+    wrapped_result = get_users(user_ids)
+    direct_result = get_users.cb(user_ids)
+    assert len(wrapped_result) == len(direct_result)
+    for wrapped, direct in zip(wrapped_result, direct_result):
+        assert wrapped == direct
+
+    with assume_test_silo_mode(SiloMode.CONTROL):
+        for user in users:
+            user.update(username=user.username + "moo")
+
+    # Does not include updates made to db
+    after_update_wrapped = get_users(user_ids)
+    for u in after_update_wrapped:
+        assert not u.username.endswith("moo")
+        # Clear cache simulating outbox logic
+        region_caching_service.clear_key(
+            region_name=get_local_region().name, key=get_users.key_from(u.id)
+        )
+
+    cached_users = get_users(user_ids)
+    for u, cached in zip(users, cached_users):
+        assert cached
+        assert cached.username.endswith("moo")
+        assert cached.username == u.username
+
+
+@django_db_all(transaction=True)
+def test_caching_many_partial() -> None:
+    cache.clear()
+
+    @back_with_silo_cache_many(base_key="get_users", silo_mode=SiloMode.REGION, t=RpcUser)
+    def get_users(user_ids: list[int]) -> list[RpcUser]:
+        return user_service.get_many(filter=dict(user_ids=user_ids))
+
+    users = [Factories.create_user() for _ in range(3)]
+    user_ids = [u.id for u in users]
+
+    single_user = get_users([users[0].id])
+    assert len(single_user) == 1
+    assert single_user[0].id == users[0].id
+    assert single_user[0].username == users[0].username
+
+    all_users = get_users(user_ids)
+    assert len(all_users) == len(users)
+    for i, user in enumerate(all_users):
+        assert user_ids[i] == user.id, "Results should be ordered based on id list"
+
+
+@django_db_all(transaction=True)
+def test_caching_many_missing_ids() -> None:
+    cache.clear()
+
+    @back_with_silo_cache_many(base_key="get_users", silo_mode=SiloMode.REGION, t=RpcUser)
```

Comment:
Would it be helpful if we assert on the contents of the cache itself within testing?